src/date/expand-pattern.js

```diff
@@ -33,10 +55,27 @@ return function( pattern, cldr ) {
 
 	switch ( true ) {
 		case "skeleton" in pattern:
+			skeleton = pattern.skeleton;
 			result = cldr.main([
 				"dates/calendars/gregorian/dateTimeFormats/availableFormats",
-				pattern.skeleton
+				skeleton
 			]);
+			if ( !result ) {
+				timeSkeleton = skeleton.split( /[^hHKkmsSAzZOvVXx]/ ).slice( -1 )[ 0 ];
+				dateSkeleton = skeleton.split( /[^GyYuUrQqMLlwWdDFgEec]/ )[ 0 ];
+				if (  /(MMMM|LLLL).*[Ec]/.test( dateSkeleton ) ) {
+					type = "full";
+				} else if ( /MMMM/g.test( dateSkeleton ) ) {
+					type = "long";
+				} else if ( /MMM/g.test( dateSkeleton ) || /LLL/g.test( dateSkeleton ) ) {
+					type = "medium";
+				} else {
+					type = "short";
+				}
+				result = getDateTime( type, dateSkeleton, timeSkeleton,
+						"dates/calendars/gregorian/dateTimeFormats/availableFormats",
+						"dates/calendars/gregorian/dateTimeFormats/availableFormats", cldr );
```

Comment:
There's something I don't like `getDateTime()`. It's not too intuitive to understand it.

Our goal here is to merge two pattern strings (e.g., `MMM y G` and `HH:mm:ss`) using a merge-pattern (e.g., `{1}, {0}`). For example, producing `MMM y G, HH:mm:ss`.

I suggest changing the code according to:

```diff
-	var result, getDateTime, skeleton, dateSkeleton = "", timeSkeleton = "", type;
+	var dateSkeleton, result, skeleton, timeSkeleton, type;
 
-	getDateTime = function( type, dateSkeleton, timeSkeleton, datePath, timePath, cldr ) {
-		var result;
-		result = cldr.main([
-			"dates/calendars/gregorian/dateTimeFormats",
-			type
-		]);
-		if ( result ) {
-			result = formatMessage( result, [
-				cldr.main([
-					timePath,
-					timeSkeleton
-				]),
-				cldr.main([
-					datePath,
-					dateSkeleton
-				])
-			]);
-		}
-		return result;
-	};
+	function combineDateTime( type, datePattern, timePattern ) {
+		return formatMessage(
+			cldr.main([
+				"dates/calendars/gregorian/dateTimeFormats",
+				type
+			]),
+			[ datePattern, timePattern ]
+		);
+	}
 
 	if ( typeof pattern === "string" ) {
 		pattern = { skeleton: pattern };
@@ -72,9 +61,16 @@ return function( pattern, cldr ) {
 				} else {
 					type = "short";
 				}
-				result = getDateTime( type, dateSkeleton, timeSkeleton,
+				result = combineDateTime( type,
+					cldr.main([
 						"dates/calendars/gregorian/dateTimeFormats/availableFormats",
-						"dates/calendars/gregorian/dateTimeFormats/availableFormats", cldr );
+						dateSkeleton
+					]),
+					cldr.main([
+						"dates/calendars/gregorian/dateTimeFormats/availableFormats",
+						timeSkeleton
+					])
+				);
 			}
 			break;
 
@@ -88,9 +84,16 @@ return function( pattern, cldr ) {
 			break;
 
 		case "datetime" in pattern:
-			result = getDateTime( pattern.datetime, pattern.datetime, pattern.datetime,
+			result = combineDateTime( pattern.datetime,
+				cldr.main([
 					"dates/calendars/gregorian/dateFormats",
-					"dates/calendars/gregorian/timeFormats", cldr );
+					pattern.datetime
+				]),
+				cldr.main([
+					"dates/calendars/gregorian/timeFormats",
+					pattern.datetime
+				])
+			);
 			break;
 
 		case "pattern" in pattern:
```